mibipret/data/data.py

```diff
@@ -0,0 +1,512 @@
+    -------
+        pandas.DataFrame: Tabular data
+        
+    Raises:
+    -------
+        ValueError: If `file_path` is not a valid file location
+
+
+    Example
+    -------
+       This function can be called with the file path of the example data as argument using
+
+        >>> from mibipret.data import load_excel
+        >>> load_excel(example_data.xlsx)
+        
+
+    """
+    # if file_path == 'nobody':
+    #     raise ValueError('Can not say hello to nobody')
+    # return f'Hello {file_path}!' 
+
+    if file_path is None:
+        raise ValueError('Specify file path and file name!')
+    if not os.path.isfile(file_path):
+        raise ValueError('Specified file does not exist')
+    
+    data = pd.read_excel(file_path, sheet_name = sheet_name,**kwargs)
+    if ";" in data.iloc[1][0]:
+        data = pd.read_excel(file_path, sep=";", sheet_name = sheet_name,**kwargs)
+    units = data.iloc[0]
+
+    return data, units
+    
+def load_csv(        
+        file_path = None,
+        verbose = False,
+        store_provenance = False,
+        ):
+    """
+    Function to load data from csv file.
+    
+    Args:
+    -------
+        file_path (str): Name of the path to the file
+        verbose (bool): flag 
+        store_provenance (bool): ...
+
+    Returns
+    -------
+        pandas.DataFrame: Tabular data
+        
+    Raises:
+    -------
+        ValueError: If `file_path` is not a valid file location
+
+
+    Example
+    -------
+       This function can be called with the file path of the example data as argument using
+
+        >>> from mibipret.data import load_excel
+        >>> load_excel(example_data.csv)
+
+
+    """
+    if file_path is None:
+        raise ValueError('Specify file path and file name!')
+    if not os.path.isfile(file_path):
+        raise ValueError('Specified file does not exist!')
+    
+    data = pd.read_csv(file_path, encoding="unicode_escape")
+    if ";" in data.iloc[1][0]:
+        data = pd.read_csv(file_path, sep=";", encoding="unicode_escape")
+    units = data.iloc[0]
+
+    if verbose:
+        print("Unit of quantities: \n", units)
+        print("Loaded data:\n", data)
+    return data, units
+
+def check_columns(data, verbose = True):
+    """
+    Function that looks at the column names and renames the columns to the standard names of the model
+    
+    Args:
+    -------
+        data (df): dataframe with the measurements
+    
+    
+    Returns
+    -------
+        pandas.DataFrame: Tabular data with standard column names
+    
+        
+    Raises:
+    -------
+    
+    Example
+    -------
+    
+    Todo's:
+        - complete list of potential contaminants, environmental factors
+        - add name check for metabolites?
+        - return column names that have not been identified
+        - option to return only columns identified
+        - add key-word to specify which data to extract (i.e. data columns to return)
+    
+    """
+  
+    col_dict = {
```

Comment:
It might be a better idea to move these kinds of dictionaries/data outside of the functions and probably even to another file. Right now it makes the function very bulky and hard to read.